### PR TITLE
Use batch inserts into CAS and RocksDbTables

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: src/Chainweb/BlockHeaderDB.hs [hunk 1]
```diff
@@ -316,9 +316,12 @@ dbAddChecked db e = unlessM (casMember (_chainDbCas db) ek) dbAddCheckedInternal
...
     -- different tables)
-    add = do
-        casInsert (_chainDbCas db) (RankedBlockHeader e)
-        tableInsert (_chainDbRankTable db) (_blockHash e) (_blockHeight e)
+    add = updateBatch
+            [ RocksDbInsert (_chainDbCas db) (casKey rbh) rbh
+            , RocksDbInsert (_chainDbRankTable db) (_blockHash e) (_blockHeight e)
+            ]
+      where
+        rbh = RankedBlockHeader e
```
Comment (fosskers) [inline src/Chainweb/BlockHeaderDB.hs]: `rbh` is only used in one place, so the `where` clause might not be necessary.

File: src/Chainweb/CutDB.hs [hunk 2]
```diff
@@ -519,11 +520,11 @@ cutHashesToBlockHeaderMap
...
     trace logfun "Chainweb.CutDB.cutHashesToBlockHeaderMap" (_cutId hs) 1 $ do
-        hdrs <- emptyCas
-        traverse_ (casInsert hdrs) $ _cutHashesHeaders hs
-
         plds <- emptyCas
-        traverse_ (casInsert plds) $ _cutHashesPayloads hs
+        casInsertBatch plds $ V.fromList $ HM.elems $ _cutHashesPayloads hs
+
+        hdrs <- emptyCas
+        casInsertBatch hdrs $ V.fromList $ HM.elems $ _cutHashesHeaders hs
```
Comment (fosskers) [inline src/Chainweb/CutDB.hs]: How many items do we expect to insert at once? Could this ever be an expensive `Vector` conversion? GHC might be nice and fuse the inner list away anyway.

Comment (larskuhtz) [reply src/Chainweb/CutDB.hs]: Agreed, using vector in the class API was probably a mistake. Lazy `[]` would have been more flexible. But changing the `IsCas` class was beyond the scope of this PR.

In any case in practice this shouldn't make a big difference. Batches shouldn't be long in any case. Not sure, though if GHC is smart enough to scope the allocation for the vector.

File: src/Chainweb/BlockHeaderDB.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/BlockHeaderDB.hs]: Actually, it's used in two places.

Comment (fosskers) [reply src/Chainweb/BlockHeaderDB.hs]: My mistake!

File: src/Chainweb/CutDB.hs (hunk 2 above)
Comment (fosskers) [reply src/Chainweb/CutDB.hs]: Usually it is in the case of pulling keys lists from a `Map`.

**Pull request merged**